components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/processors/RequestHandler.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.gateway.handlers.analytics.processors;
+
+import org.apache.synapse.MessageContext;
+
+/**
+ * interface to request handlers
+ */
+public interface RequestHandler {
+    void handleRequest(MessageContext messageContext);
```

Comment:
Since what this does is extracting details from the message context and publishing event through the publishing client, we might have to use a better name for the Interface and for the method too. Something like AnalyticsEventHandler as the interface name and convertAndPublish as the method name. The name handleRequest gave an idea that this is getting invoked in the request path...